www/src/components/chat/chat-list.tsx

```diff
@@ -94,25 +91,34 @@ export const ChatList: React.FC = React.memo(() => {
   ]);
 
   const handleCopyMessage = useCallback(
-    (message: string) => {
-      navigator.clipboard.writeText(message).then(() => {
+    (message: Message) => {
+      const textToCopy = isStructuredMessage(message)
+        ? message.message.answer
+        : message.message;
+      navigator.clipboard.writeText(textToCopy).then(() => {
         toast({
           title: "Copied to clipboard",
           description: "The message has been copied to your clipboard.",
         });
       });
     },
-    [toast],
+    [toast]
   );
 
   const handleOnClickEditMessage = useCallback((message: Message) => {
-    setEditMessageContent(message.message);
+    setEditMessageContent(
+      isStructuredMessage(message) ? message.message.answer : message.message
+    );
     setIsEditable(message.id);
   }, []);
 
-  const messageContent = useCallback((messageText: string) => {
-    const formattedText = formatTextWithReferences(messageText);
-    return formattedText.replace(/\n/g, "<br />");
+  const messageContent = useCallback((message: Message) => {
+    if (!isStructuredMessage(message)) return message.message;
+    if (isStructuredMessage(message) && !message.message.url_supporting.length) return message.message.answer;
```

Comment:
these checks are looking a bit weird to me -- should we add some docs/comments about why this happens?

Reply:
@ribeirojose  Maybe if I change the name from "isStructuredMessage" to something like "isChatAnswer" do you think that It would make more sense and easier to read?

Reply:
Or maybe some better name too

Reply:
no, the issue is I don't understand why/it doesn't make much sense to me why this could have two formats